src/main/java/uk/gov/ons/census/casesvc/messaging/RmNonComplianceReceiver.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.ons.census.casesvc.messaging;
+
+import static uk.gov.ons.census.casesvc.utility.MsgDateHelper.getMsgTimeStamp;
+
+import java.time.OffsetDateTime;
+import org.springframework.integration.annotation.MessageEndpoint;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.messaging.Message;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.StringUtils;
+import uk.gov.ons.census.casesvc.logging.EventLogger;
+import uk.gov.ons.census.casesvc.model.dto.CollectionCase;
+import uk.gov.ons.census.casesvc.model.dto.ResponseManagementEvent;
+import uk.gov.ons.census.casesvc.model.entity.Case;
+import uk.gov.ons.census.casesvc.model.entity.CaseMetadata;
+import uk.gov.ons.census.casesvc.model.entity.EventType;
+import uk.gov.ons.census.casesvc.service.CaseService;
+
+@MessageEndpoint
+public class RmNonComplianceReceiver {
+
+  private final CaseService caseService;
+  private final EventLogger eventLogger;
+
+  public RmNonComplianceReceiver(CaseService caseService, EventLogger eventLogger) {
+    this.caseService = caseService;
+    this.eventLogger = eventLogger;
+  }
+
+  @Transactional
+  @ServiceActivator(inputChannel = "rmNonComplianceInputChannel")
+  public void receiveMessage(Message<ResponseManagementEvent> message) {
+    OffsetDateTime messageTimestamp = getMsgTimeStamp(message);
+    ResponseManagementEvent responseManagementEvent = message.getPayload();
+    CollectionCase collectionCase = responseManagementEvent.getPayload().getCollectionCase();
+
+    Case caze = caseService.getCaseByCaseId(collectionCase.getId());
+    caze.setMetadata(getCaseMetaData(caze));
+    caze.getMetadata().setNonCompliance(collectionCase.getNonComplianceStatus());
+
+    if (!StringUtils.isEmpty(collectionCase.getFieldOfficerId())) {
+      caze.setFieldOfficerId(collectionCase.getFieldOfficerId());
+    }
+
+    if (!StringUtils.isEmpty(collectionCase.getFieldCoordinatorId())) {
+      caze.setFieldCoordinatorId(collectionCase.getFieldCoordinatorId());
+    }
+
+    caseService.saveCaseAndEmitCaseUpdatedEvent(caze, null);
```

Comment:
if the case was marked for fieldwork, it isn't sent to Fieldwork Adapter as a CREATE or whatever?

Reply:
I think this is covered by a followup ticket: https://trello.com/c/X9hDP5I3/1379-nc-import-file-2-and-send-cases-to-field-8